gnosisguild/grounds-monorepo
# Feature: Implement Wallet Connection Modal with Connection Logic

## Overview
This pull request introduces a series of updates to the wallet connection features of our React application. The changes include UI components, state management enhancements, and connection logic for different wallet providers.

## Changes

### UI Components
- **Header Component**: Updated the header layout to include the logo and a 'Connect' button, aligning them using Material-UI's `Stack` component with `justifyContent` set to `space-between`.
- **WalletModal Component**: Created a new modal component using Material-UI's `Modal`, `Box`, `Typography`, `Button`, `IconButton`, and `Grid` components to allow users to choose from various wallet connection options.

### Connection Logic
- Integrated the logic for activating and deactivating wallet connections within the `WalletModal` component. This includes support for MetaMask, WalletConnect, and Coinbase Wallet.
- Included checks for MetaMask extension installation and provided a fallback option to direct users to the MetaMask installation page if it's not detected.
- Added state management to track the active connection and update the UI accordingly, disabling the connect button for the active wallet.

### Accessibility and Responsiveness
- Ensured that the wallet connection modal is accessible and responsive, with consideration for screen readers and various device sizes.

## How to Test
1. Open the application and click on the 'Connect' button in the header.
2. In the modal, choose a wallet provider and attempt to connect.
3. Check the connection status indicated by the UI.
4. Attempt to disconnect the active wallet connection.
5. Confirm that the WalletModal appears on top of the WalletConnect modal when opened.


Please review the changes and provide feedback or approval for merging this feature into the main branch.
